frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-topology/WhenExpressionForm.tsx

```diff
@@ -0,0 +1,63 @@
+import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+import { useFormikContext } from 'formik';
+import { Grid, GridItem, TextInputTypes } from '@patternfly/react-core';
+import { InputField, DropdownField, TextColumnField } from '@console/shared';
+import AutoCompletePopover from '../../shared/common/auto-complete/AutoCompletePopover';
+import { PipelineBuilderFormikValues } from '../pipeline-builder/types';
+import { WhenExpressionOperatorType } from '../pipeline-builder/const';
+
+type WhenExpressionFormProps = {
+  namePrefix?: string;
+  autoCompleteValues: string[];
+};
+
+const WhenExpressionForm: React.FC<WhenExpressionFormProps> = ({
+  namePrefix,
+  autoCompleteValues,
+}) => {
+  const { setFieldValue } = useFormikContext<PipelineBuilderFormikValues>();
+  const { t } = useTranslation();
+
+  return (
+    <Grid hasGutter>
+      <GridItem span={6}>
+        <AutoCompletePopover
+          autoCompleteValues={autoCompleteValues}
+          onAutoComplete={(input: string) => setFieldValue(`${namePrefix}.input`, input)}
+        >
+          {(ref) => (
+            <InputField
+              ref={ref}
+              type={TextInputTypes.text}
+              name={`${namePrefix}.input`}
+              label={t('pipelines-plugin~Input')}
+              autocomplete="off"
+            />
+          )}
+        </AutoCompletePopover>
+      </GridItem>
+      <GridItem span={6}>
+        <DropdownField
+          name={`${namePrefix}.operator`}
+          label={t('pipelines-plugin~Operator')}
+          items={WhenExpressionOperatorType}
+          title={t('pipelines-plugin~Select Operator')}
+          onChange={(operator: string) => setFieldValue(`${namePrefix}.operator`, operator)}
+          fullWidth
+        />
+      </GridItem>
+      <GridItem>
+        <TextColumnField
```

Comment:
FWIW, I need to do this for my PR https://github.com/openshift/console/pull/8752